litr-demo/src/main/java/com/linkedin/android/litr/demo/RecordCameraFragment.kt

```diff
@@ -0,0 +1,358 @@
+        }
+
+        val recordSize = getRecordSize(sensorRect)
+        binding.videoMediaSource?.apply {
+            width = recordSize.width
+            height = recordSize.height
+            bitrate = DEFAULT_TARGET_BITRATE
+            frameRate = DEFAULT_CAMERA_FPS
+            orientation = 0
+        }
+
+        binding.transformationPresenter?.recordCamera(
+                binding.audioMediaSource!!,
+                binding.videoMediaSource!!,
+                binding.targetMedia!!,
+                binding.transformationState!!
+        )
+    }
+
+    private fun startCapture(camera: CameraDevice) {
+        val surfaces = mutableListOf<Surface>()
+        captureRequest = camera.createCaptureRequest(CameraDevice.TEMPLATE_RECORD).apply {
+            surfaceHolder?.let {
+                addTarget(it.surface)
+                surfaces.add(it.surface)
+            }
+            transformerTexture?.let {
+                addTarget(it)
+                surfaces.add(it)
+            }
+
+            // Set a fixed target frame rate. This will match what we'll assume we are encoding too.
+            set(CaptureRequest.CONTROL_AE_TARGET_FPS_RANGE,
+                    Range(DEFAULT_CAMERA_FPS, DEFAULT_CAMERA_FPS))
+        }.build()
+
+        camera.createCaptureSession(surfaces, captureStateCallback, backgroundHandler)
+    }
+
+    private fun stopRecording() {
+        binding.transformationPresenter?.stopRecording(
+                binding.audioMediaSource!!,
+                binding.videoMediaSource!!
+        )
+
+        captureSession?.apply {
+            stopRepeating()
+            close()
+        }
+    }
+
+    private fun startBackgroundThread() {
+        backgroundHandlerThread = HandlerThread("CameraVideoThread")
+        backgroundHandlerThread.start()
+        backgroundHandler = Handler(backgroundHandlerThread.looper)
+    }
+
+    private fun stopBackgroundThread() {
+        backgroundHandlerThread.quitSafely()
+        backgroundHandlerThread.join()
+    }
+
+    private fun hasAudioRecordPermission(): Boolean {
+        val validContext = context ?: return false
+        return ContextCompat.checkSelfPermission(
+                validContext,
+                Manifest.permission.RECORD_AUDIO
+        ) == PackageManager.PERMISSION_GRANTED
+    }
+
+    private fun hasCameraPermission(): Boolean {
+        val validContext = context ?: return false
+        return ContextCompat.checkSelfPermission(
+                validContext,
+                Manifest.permission.CAMERA
+        ) == PackageManager.PERMISSION_GRANTED
+    }
+
+    /**
+     * Since we don't support cropping of the sensor or input surface, we will record to a Surface
+     * that has the same aspect ratio as the camera's sensor. We will target something similar to
+     * 720p (1280 x 720) but allow the height to increase when required for non-16:9 ratios.
+     */
+    private fun getRecordSize(sensorRect: Rect?): Size {
```

Comment:
For now, we will require the source aspect ratio to match the target aspect ratio. For the camera specific cases, we could follow up with a couple of options:
 - Have people crop at the sensor
 - Implement some filter support for cropping

While the former is arguably simpler, it introduces the limitation that if the application is using a Preview Surface, then it will also be cropped. A crop based filter may be a better / more versatile solution.